shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java

```diff
@@ -281,22 +285,32 @@ public Context createDisplayContext(Display display) {
     /**
      * Proxies a Context replacing the WindowManager and InputMethodManager with our custom instance.
      */
-    private static class PresentationContext extends ImmContext {
+    // TODO(mklim): This caches the IMM at construction time and won't pick up any changes. In rare
+    // cases where the FlutterView changes windows this will return an outdated instance. This
+    // should be fixed to instead defer returning the IMM to something that know's FlutterView's
+    // true Context.
+    private static class PresentationContext extends ContextWrapper {
         private @NonNull
         final WindowManagerHandler windowManagerHandler;
         private @Nullable
         WindowManager windowManager;
+        private @NonNull
+        final InputMethodManager inputMethodManager;
 
         PresentationContext(Context base, @NonNull WindowManagerHandler windowManagerHandler) {
             super(base);
             this.windowManagerHandler = windowManagerHandler;
+            this.inputMethodManager = (InputMethodManager) base.getSystemService(INPUT_METHOD_SERVICE);
         }
 
         @Override
         public Object getSystemService(String name) {
             if (WINDOW_SERVICE.equals(name)) {
                 return getWindowManager();
             }
+            if (INPUT_METHOD_SERVICE.equals(name)) {
```

Comment:
Do we need this? (I thought the base context would be based of an ImmContext anyway)

Reply:
:man_facepalming: We don't need it, thanks. Removed.